packages/fxa-content-server/app/scripts/views/settings/display_name.jsx

```diff
@@ -0,0 +1,203 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import AvatarMixin from '../mixins/avatar-mixin';
+import Cocktail from 'cocktail';
+import BaseView from '../base';
+import SettingsPanelMixin from '../mixins/settings-panel-mixin';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Translator from '../../lib/translator';
+
+const translator = new Translator();
+
+const t = msg => translator.get(msg);
+
+function DisplayName(props) {
+  return (
+    <div id="display-name" className="settings-unit">
+      <div className="settings-unit-stub">
+        <DisplayNameComponent />
+        <ChangeorAddButtonComponent displayName={props.displayName} />
+      </div>
+      <div className="settings-unit-details">
+        <div className="error" />
+        <DisplayNameFormComponent
+          account={props.account}
+          submit={props.submit}
+          displayName={props.displayName}
+        />
+      </div>
+    </div>
+  );
+}
+function DisplayNameComponent() {
+  return (
+    <header className="settings-unit-summary">
+      <h2 className="settings-unit-title">{t('Display name')}</h2>
+    </header>
+  );
+}
+function ChangeorAddButtonComponent(props) {
```

Comment:
For example, if this component were `export`ed, you could write a small test for `ChangeorAddButtonComponent` in isolation that exercises what it does with and without `displayName` defined.

You might also be able to separate React tests from the rest of the existing tests. I have a suspicion that might make things simpler.